kart/diff_structs.py

```diff
@@ -430,6 +433,10 @@ def __json__(self):
         return {k: v for k, v in self.items()}
 
 
+class InvalidatedDeltaDiff(Exception):
+    pass
+
+
 class DeltaDiff(Diff):
     """
```

Comment:
You've made some big changes to DeltaDiff - add something to the docstring

Need to briefly document what items, iter_items, data, _lazy_initial_contents, and _evaluate_lazy_initial_contents is / how they fit together.